staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go

```diff
@@ -32,6 +32,104 @@ import (
+			trigger: func(gvr schema.GroupVersionResource, ns string, fakeClient *fake.FakeDynamicClient, _ *unstructured.Unstructured) *unstructured.Unstructured {
+				testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
+				createdObj, err := fakeClient.Resource(gvr).Namespace(ns).Create(testObject, metav1.CreateOptions{})
+				if err != nil {
+					t.Error(err)
+				}
+				return createdObj
+			},
+			handler: func(rcvCh chan<- *unstructured.Unstructured) *cache.ResourceEventHandlerFuncs {
+				return &cache.ResourceEventHandlerFuncs{
+					AddFunc: func(obj interface{}) {
+						rcvCh <- obj.(*unstructured.Unstructured)
+					},
+				}
+			},
+		},
+		// scenario 2
+		{
+			name:     "scenario 2: test adding an object should trigger AddFunc",
+			informNS: "ns-foo",
+			ns:       "ns-foo",
+			gvr:      schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
+			trigger: func(gvr schema.GroupVersionResource, ns string, fakeClient *fake.FakeDynamicClient, _ *unstructured.Unstructured) *unstructured.Unstructured {
+				testObject := newUnstructured("extensions/v1beta1", "Deployment", "ns-foo", "name-foo")
+				createdObj, err := fakeClient.Resource(gvr).Namespace(ns).Create(testObject, metav1.CreateOptions{})
+				if err != nil {
+					t.Error(err)
+				}
+				return createdObj
+			},
+			handler: func(rcvCh chan<- *unstructured.Unstructured) *cache.ResourceEventHandlerFuncs {
+				return &cache.ResourceEventHandlerFuncs{
+					AddFunc: func(obj interface{}) {
+						rcvCh <- obj.(*unstructured.Unstructured)
+					},
+				}
+			},
+		},
+	}
+
+	for _, ts := range scenarios {
+		t.Run(ts.name, func(t *testing.T) {
+			// test data
+			timeout := time.Duration(3 * time.Second)
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
+			defer cancel()
+			scheme := runtime.NewScheme()
+			informerReciveObjectCh := make(chan *unstructured.Unstructured, 1)
+			objs := []runtime.Object{}
+			if ts.existingObj != nil {
+				objs = append(objs, ts.existingObj)
+			}
+			fakeClient := fake.NewSimpleDynamicClient(scheme, objs...)
+			target := dynamicinformer.NewFilteredDynamicSharedInformerFactory(fakeClient, 0, ts.informNS, nil)
+
+			// act
+			informerListerForGvr := target.ForResource(ts.gvr)
+			informerListerForGvr.Informer().AddEventHandler(ts.handler(informerReciveObjectCh))
+			target.Start(ctx.Done())
+			if synced := target.WaitForCacheSync(ctx.Done()); !synced[ts.gvr] {
+				t.Errorf("informer for %s hasn't synced", ts.gvr)
+			}
+
+			testObject := ts.trigger(ts.gvr, ts.ns, fakeClient, ts.existingObj)
+			select {
+			case objFromInformer := <-informerReciveObjectCh:
+				if ts.ns != ts.informNS {
+					t.Errorf("informer received an object for namespace %s when watching namespace %s", ts.ns, ts.informNS)
+				}
+				if !equality.Semantic.DeepEqual(testObject, objFromInformer) {
+					t.Fatalf("%v", diff.ObjectDiff(testObject, objFromInformer))
```

Comment:
nit: what's difference between `t.Errorf` and `t.Fatalf` here?

Reply:
Honestly, I don't know why `t.Fatalf` is used there. I copied the structure from the other dynamicinformer test and that is what is does. I am happy to change it either way.